panel/lxqtpanel.cpp

```diff
@@ -230,6 +233,30 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
 
     loadPlugins();
 
+    if(qGuiApp->nativeInterface<QNativeInterface::QWaylandApplication>())
+    {
+        // Create backing QWindow for LayerShellQt integration
+        create();
```

Comment:
Hi. Good question...
I've found `create()` in [dprgadget example](https://github.com/qt/qtbase/blob/ae361f00b3f4827e62a19cc451227c9892b51f2e/tests/manual/highdpi/dprgadget/main.cpp#L197), so I've used that.
`winId()` call seems to differ only in forcing native widgets if requested.

About show/hide this is not a problem. `QWindow` is hidden but not deleted so no need to manually re-attach the `LayerShellQt` integration. It will automatically apply to the new Wayland surface.

Reply:
> About show/hide this is not a problem. QWindow is hidden but not deleted so no need to

That sounds logical, but I remember that we needed to do it before showing the widget in lxqt-runner and lxqt-notificationd. However, the reason may be in the differences between the apps; I also did in inside the c-tor in pcmanfm-qt's desktop, like you did here.

As for `create()`, looking at its code in Qt source, I thought it might be too much here.

Reply:
Current code works as is but both `winId()` and `create()` are ok for me.

Reply:
> both winId() and create() are ok for me

For me too. Let's keep `create()` here to see what will happen. It was OK in my tests.